doc/user-manual/getting-started/hello-world.lagda.rst

```diff
@@ -7,35 +7,417 @@
+could have been ``m`` and so on).
+
+We refer to ``A`` as the *parameter* of the datatype ``Vec A n``. So far, we
+know that the index ``n`` is a natural number, but we also need to know what
+is ``A``. We will specify that A can be any element type of the sort ``Set``,
+but it could have been any other sort with a greater level of abstraction. We
+express the fact that ``A`` has type ``Set`` or ``A`` ranges over ``Set`` by
+means of the colon ``:``.
+
+We can infer that ``Vec A n`` will also range over ``Set`` (the lower upper
+bound of the types involved). We are now entitled to write the signature of
+our datatype: ``Vec (A : Set) : ℕ → Set``.
+
+To declare the datytype, we place its signature between the keywords ``data``
+and ``where``.
+
+The constructors ``[]`` and ``_∷_``
+-----------------------------------
+
+Constructors are declared in new lines and indented with two spaces.
+
+We chose for the first constructor the identifier ``[]``. It represents the
+empty vector, and its type is the vector space of length ``0``.
+
+The second constructor is a :ref:`mixfix operator <mixfix-operators>` named
+``_∷_`` (pronounced append). For all numbers, it takes as input an element
+of ``A`` and a vector. As output, it produces a vector with a length
+increased by one.
+
+The declaration with keyword ``ìnfixr`` does not belong to the datatype
+declaration; therefore it is not indented. It establishes the
+:ref:`precedence <precedence>` of the operator append. This finishes our
+explanation of the ‘Hello world’ dependent type example.
+
+‘Hello world’ proof
+===================
+
+We will see in this section how to enunciate and prove with Agda the
+associativity of natural numbers under the addition. This time we will build
+the code line by line. You can load the file in each step.
+
+Enunciation of associativity
+----------------------------
+
+Please create a new ``.agda`` file named ``hello-world-proof.agda``.
+Paste or type the following code:
+
+.. code-block:: agda
+
+  module hello-world-proof where
+
+Loading the file with ``C-c C-l`` should work. This is fon now the empty
+module. Now we bring to scope the datatype ``Nat`` and the addition operation
+``_+_``, both defined in the Agda Builtin library.
+
+.. code-block:: agda
+
+  open import Agda.Builtin.Nat using (Nat ; _+_)
+
+Next, we import the propositional equality between two terms ``_≡_``. This will
+allow us to state that two terms compute to the same normal form.
+
+.. code-block:: agda
+
+  open import Agda.Builtin.Equality using (_≡_)
+
+Now we can enunciate the associativity: given three (possibly different)
+natural numbers, adding the first to the addition of the second and the third
+computes to the same value as adding the addition of the first and the second
+to the third. We will name this statement ``+assoc-enun``.
+
+.. code-block:: agda
+
+  +assoc-enun : Set
+  +assoc-enun = ∀ (x y z : Nat) → x + (y + z) ≡ (x + y) + z
```

Comment:
It's not very common to explicitly define the statement of a proposition as an element of type `Set`, so I don't think it should be included here.

Reply:
You are right; it is not very common. But I still think it is good to separate the statement lemma from its proof. Sometimes, the formalization of the statement of a theorem is hard enough and itself a result. This gives us also the opportunity of remarking in which universe the statement lives. Also that finding its proof is just finding an element of that type.